python/cudf/cudf/core/column/struct.py

```diff
@@ -191,15 +191,21 @@ def field(self, key):
             pos = fields.index(key)
             return self._return_or_inplace(self._column.children[pos])
         else:
-            try:
-                return self._return_or_inplace(self._column.children[key])
-            except Exception:
-                print(
+            if isinstance(key, int):
+                try:
+                    return self._return_or_inplace(self._column.children[key])
+                except IndexError:
+                    error_message_string = (
+                        "Index " + str(key) + " out of range"
+                    )
+                    raise IndexError(error_message_string)
```

Comment:
An f-string could help make things more concise here; ditto below:

```
IndexError(f"Index {key} out of range")
```

Otherwise, the new changes look great!

Reply:
Great idea;  thank you @shwina!!